### PR TITLE
Bug fixed in the GraphQL resolver

### PR DESCRIPTION
**Pull request by olliefr**
## Update

As discussed, this is useful enough to warrant inclusion into `main`. The siblings bug described below had been fixed by @jlarfors, and I have added extra tests for the GraphQL resolver. 

These things lay the foundation for moving forward with the upcoming enhancements to the resolver, as well as facilitate making progress with the pilots :shipit: 

## Original issue

I could use some help with this. There are two issues. I've made some changes to the code, but the issues were there to begin with, that is the `main` branch has them too, it's just we didn't test for these before.

The main change I made is to remove node hopping. We can't skip nodes because in a schema, where multiple routes exist between nodes, the path between two nodes is not uniquely defined.

I've also added a bunch of tests with smaller schema/data file in hopes to isolate the problems better and to set up the foundation for adding more test cases without having to clutter _THE_ one and only test schema. We'll need more test cases for the GraphQL arguments, so this is a good start.

You don't have to run the whole test suite to see the issues in action, one of `TestPostgres` or `TestMe`, or `TestPostgresSQLGen` in `store` module will do. `TestMe` is meant to be a temporary short(er) test for debug purposes.

## Inverse relationships

The GraphQL resolver fails to correctly resolve inverse relationships, such as the following.

```hcl
# The parent table
table "A" {

  # field whatever a

  # The related table
  table "B" {
     # field whatever b
  }
}
```

In this case, the following GraphQL query would work.

```graphql
A {
  # field whatever a

  B {
     # field whatever b
  }
}
```

But the inverse query won't.

```graphql
B {
   # field whatever b

  A {
    # field whatever A
  }
}
```

In this case, `B` will point to a `nil`. I believe the relationship should work both ways.

## Sibling fields.

Now consider the schema

```hcl
# The parent table
table "A" {

  # field whatever a

  # First related table
  table "B" {
    # field whatever b
  }

  # Second related table
  table "C" {
    # field whatever c
  }
}
```

With this, either of the following GraphQL queries would succeed

```graphql
A {
  # field whatever a
  B {
    # field whatever b
  }
}
```

```graphql
A {
  # field whatever a
  C {
    # field whatever c
  }
}
```

But the field with the sibling related tables would fail, the second table would return null

```graphql
A {
  # field whatever a
  B {
    # field whatever b
  }
  C {
    # field whatever c
  }
}
```

The odd thing about this is that the correct SQL is being generated, and the correct result is returned from the database, as far as I could tell. But somehow GraphQL module ignores the second sibling element.

Helb?! 🙀

### CODE REVIEW

File: store/schemagraph.go [hunk 1]
```diff
@@ -6,16 +6,50 @@ import (
 	"github.com/valocode/bubbly/api/core"
 )
 
-// schemaGraph creates a graph from the bubbly schema
+// relType describes the relationship type of a directed edge from a --> b
+type relType int
+
+// FIXME: ofr what's the difference between `oneToOne` and `belongsTo`, please?
```
Comment (jlarfors) [inline store/schemagraph.go:12]: The order.

```hcl
table "A" {
  table "B" { unique = true }
}
```

Table B belongs to A. And Table A has a oneToOne to B. So the relationships describe the direction of the edge.

File: store/schemagraph.go [hunk 2]
```diff
@@ -6,16 +6,50 @@ import (
...
+	nodes schemaNodes
 	// nodeIndex stores an index to the nodes using the schema table name.
 	// This is probably not the best for performance, but our schemas probably
 	// will not become huge and this is a great convienice for consumers of the
 	// graph, to not have to traverse the graph to find a node
-	nodeIndex map[string]*schemaNode
+	nodeIndex nodeRefMap
 }
 
+// traverse ???
```
Comment (jlarfors) [inline store/schemagraph.go:52]: traverse traverses the schemaGraph and calls the callback function for each node

File: store/schemagraph.go [hunk 3]
```diff
@@ -26,6 +60,7 @@ func (g *schemaGraph) traverse(fnVisit func(node *schemaNode) error) error {
 	return nil
 }
 
+// visitSchemaNode ???
```
Comment (jlarfors) [inline store/schemagraph.go:63]: visitSchemaNode is used by traverse to make sure a node is only "visited" once (the callback function is called only once for each node)

**Review by benmarsden — APPROVED** [review]
LGTM. Nice work!

**Review by nate-droid — APPROVED** [review]
Awesome work!

**Pull request merged**